### DISCUSSION TITLE
Is it possible to push profiles to Phlare?

### DISCUSSION THREAD

**Discussion by prati0100** [grafana/phlare#582] [General]
Looking at the docs I have managed to get the profile scraper working. But I am looking at a large scale system where different machines will run the profiler at different times. So they will not always have any profiles to send to Phlare. The scraper requesting profiles at fixed intervals creates unnecessary network traffic.

Is it possible for machines to push the profiles to Phlare instead? If yes, how do I do that? If no, is this feature planned in the near future?

**Comment by cyriltovena** [comment]
Do you mean batching profiles before pushing them ? If so we're working on this and it will be done in the Grafana Agent.

Otherwise you can always use all the pyroscope SDK now in your applications that directly pushes
